src/Lib.hs

```diff
@@ -72,3 +77,11 @@ startServer :: IO ()
 startServer = do
   putStrLn $ "Server started at http://localhost:" ++ show portNumber
   run portNumber app
+
+makeRequest :: String -> IO (Either SomeException B.ByteString)
+makeRequest url =
+      do
+      res <- try $ get url
+      return $ case res of 
+          Right response -> Right(response ^. responseBody)
+          Left e -> Left e
```

Comment:
Pattern matching is unnecessary here, because you can just use `fmap` to transform contents of `Right`.
I will refactor this piece in next PR (without changing the behavior).
Resulting code could be:
```hs
makeRequest :: String -> IO (Either SomeException B.ByteString)
makeRequest = fmap (fmap (^. responseBody)) . try . get
```